.github/workflows/deploy-to-eks.yaml

```diff
@@ -0,0 +1,59 @@
+# USAGE
+# -----
+# # NOTE: Can be used as part of other workflows, such as ci-ecr.yaml
+# on:
+#   workflow_dispatch:
+#     branches:
+#       - main
+#   push:
+#     branches:
+#       - main
+#     paths-ignore:
+#       - "Jenkinsfile"
+#       - ".git**"
+#
+# jobs:
+#   deploy-to-eks:
+#     uses: alphagov/govuk-infrastructure/.github/workflows/deploy-to-eks.yaml@main
+#     secrets:
+#       CI_GITHUB_API_TOKEN: ${{ secrets.CI_GITHUB_API_TOKEN }}
+
+# REUSABLE WORKFLOW
+# -----------------
+
+name: Deploy to EKS
+
+on:
+  workflow_call:
+    secrets:
+      CI_GITHUB_API_TOKEN: ${{ secrets.CI_GITHUB_API_TOKEN }}
+
+jobs:
+  deploy:
+    name: Update image tag
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        env:
+          GIT_USER_EMAIL: govuk-ci@digital.cabinet-office.gov.uk
+          GIT_USER_NAME: govuk-ci
+          GITHUB_TOKEN: ${{ secrets.GOVUK_CI_GITHUB_API_TOKEN }}
+          GOVUK_ENV: integration
+          IMAGE_TAG: ${{ github.sha }}
+          REPOSITORY: ${{ github.event.repository.name }}
+        run: |
+          set -eu
+          git config --global user.email "${GIT_USER_EMAIL}"
+          git config --global user.name "${GIT_USER_NAME}"
+          echo "Modifying Helm Charts repo..."
+          repo="https://${GIT_USER_NAME}:${GITHUB_TOKEN}@github.com/alphagov/govuk-helm-charts.git"
+          branch="main"
+          file="charts/argocd-apps/image-tags/${GOVUK_ENV}/${REPOSITORY}"
+          git clone --single-branch --depth 1 --branch main $repo
+          cd "govuk-helm-charts"
+          echo $IMAGE_TAG > $file
+          git add $file
+          git commit -m "Deploy ${REPOSITORY}:${IMAGE_TAG} to ${GOVUK_ENV}"
+          REALLY_PUSH_TO_MAIN=1 git push $repo $branch
```

Comment:
> if we merge a PR after the image tag update process has made a commit, but before it's pushed, then the push will be rejected

I agree this is a risk. We've split the image tags out into individual files, so the chance of a merge conflict should be reduced: https://github.com/alphagov/govuk-helm-charts/blob/main/charts/argocd-apps/image-tags/integration/signon-resources.

Definitely worth thinking about introducing some retry behaviour if we see this error.

Reply:
Yup, think we should be okay for merge conflicts. To be clear the contention is when the remote branch (GitHub's main) has newer commits compared to the local branch. So when you go to push, conflict or not, git will reject the push because the local branch isn't up to date. But for now probably not an issue, something we can look at improving.